tape/src/main/java/com/squareup/tape/FileObjectQueue.java

```diff
@@ -75,6 +85,18 @@ public FileObjectQueue(File file, Converter<T> converter) throws IOException {
       queueFile.forEach(new QueueFile.ElementVisitor() {
         int count;
         @Override public boolean read(InputStream in, int length) throws IOException {
+          final int fileLength = queueFile.fileLength - QueueFile.HEADER_LENGTH;
+          if (length > fileLength || length > queueFile.maxElementSize) {
```

Comment:
I'm confused, why throw here instead of earlier right where we read the entry size?

Reply:
In readElement()? - we could throw there but then we would also be throwing in the header and we would need to think of any other places we would have to handle that. Perhaps best though to catch it as early as possible as you say.

Reply:
I moved it to readElement in QueueFile so it will always check.